test/jruby/test_higher_javasupport.rb

```diff
@@ -1737,6 +1737,29 @@ def test_indexed_bean_style_accessors_are_not_aliased
     end
   end
 
+  def test_no_warnings_on_interface_impls_being_set_as_constants
```

Comment:
Just to confirm, you were able to make this fail? I would have thought a more aggressive looping test would be necessary.

Reply:
yes, it does reproduce fairly reliably, at least for me locally:
```
Failure: test_no_warnings_on_interface_impls_being_set_as_constants(TestHigherJavasupport):
  /home/kares/workspace/oss/jruby/test/jruby/test_higher_javasupport.rb:1754: warning: already initialized constant org.jruby.gen::InterfaceImpl776208565
  /home/kares/workspace/oss/jruby/test/jruby/test_higher_javasupport.rb:1755: warning: already initialized constant org.jruby.gen::InterfaceImpl776208565
  .
  <85> is neither nil or false.
/home/kares/workspace/oss/jruby/test/jruby/test_higher_javasupport.rb:1760:in `test_no_warnings_on_interface_impls_being_set_as_constants'
     1757:       threads.each(&:join)
     1758:     end
     1759:     # expect no warning: already initialized constant org.jruby.gen::InterfaceImpl1353309827
  => 1760:     refute output.index('already initialized constant'), output

```